returnn/torch/distributed.py

```diff
@@ -115,7 +125,12 @@ def maybe_make_distributed_module(self, module: torch.nn.Module) -> Optional[Dis
 
     def step_after_param_update(self, *, module: torch.nn.Module, epoch_step_idx: int):
         """one train step"""
-        if self._reduce_type == "param" and ((epoch_step_idx % self._param_sync_step) == (self._param_sync_step - 1)):
+        if self._reduce_type == "custom_step_after_param_update":
+            with torch.no_grad():  # TODO: do we want this for all syncers?
+                self._custom_step_after_param_update(
+                    module=module, train_step_idx=epoch_step_idx, **get_fwd_compat_kwargs()
+                )
```

Comment:
There should be no TODO here. Either do it always (if we maybe want to remove it add some point, it means it needs another option, which we then can create later), or not at all (then it's just the user responsibility to do it inside its custom step func).
```suggestion
            self._custom_step_after_param_update(
                module=module, train_step_idx=epoch_step_idx, **get_fwd_compat_kwargs()
            )
```